src/workerd/api/http.c++

```diff
@@ -2032,6 +2032,12 @@ jsg::Promise<jsg::Ref<Response>> fetchImplNoOutputLock(jsg::Lock& js,
   //   requires a significant rewrite of the code below. It'll probably get simpler, though?
   kj::Own<kj::HttpClient> client = asHttpClient(kj::mv(clientWithTracing.client));
 
+  if (util::Autogate::Autogate::isEnabled(util::AutogateKey::FETCH_REQUEST_MEMORY_ADJUSTMENT)) {
+    // fetch requests use a lot of unaccounted c++ memory, so we simply adjust memory usage by some
+    // arbitrary amount to protect against OOMs.
+    client = client.attach(js.getExternalMemoryAdjustment(3 * 1024));
```

Comment:
Keep in mind that the memory adjustment stuff caries it's own special amount of overhead due to the cross thread signaling mechanism that's there. Be sure to benchmark this carefully.

Reply:
well, this is fetch, so it is already presumed to be slow. This can't be too bad.

Reply:
heh, you might be surprised ;-) ... but yeah, I think with all the external memory adjustment we just need to be careful.